plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt

```diff
@@ -237,17 +308,35 @@ class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
         }
       }
     }
+    if (skipRedundantAnimator(targets, cameraAnimator.type)) {
+      if (debugMode) {
+        logI(
+          TAG,
+          "Animation ${cameraAnimator.type.name}(${cameraAnimator.hashCode()}) was skipped."
+        )
+      }
+      return false
+    }
     cameraAnimator.setObjectValues(*targets)
     return true
   }
 
-  internal fun notifyListeners(cameraState: CameraState) {
-    cameraState.also {
-      bearing = it.bearing
-      center = it.center
-      padding = it.padding
-      pitch = it.pitch
-      zoom = it.zoom
+  private fun skipRedundantAnimator(targets: Array<out Any?>, type: CameraAnimatorType): Boolean {
+    if (targets.size <= 1) {
```

Comment:
Wouldn't it be faster to check if `cameraAnimator.targets.size` is 0 at the start of fun `updateAnimatorValues` instead? we would avoid reaching this point

Reply:
Moreover, we wouldn't need the below `return when (type) {...}`, would we?

Reply:
I'd simplified code a bit however, moved check for the empty target values to the very top of the method, PTAL

Reply:
Thanks for explaining!